packages/client/components/VideoControls.tsx

```diff
@@ -45,7 +45,10 @@ const VideoControls = (props: Props) => {
   })
 
   const addMedia = async () => {
-    const descriptors = ['camera', 'microphone'] as PermissionName[]
+    const descriptors = ['camera', 'microphone'] as Extract<
+      'camera' | 'microphone',
+      PermissionName
+    >[]
```

Comment:
This previously returned the error:
```
Conversion of type '("camera" | "microphone")[]' to type 'PermissionName[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```
I think `Extract` is cleaner than casting it to `unknown` and then to `PermissionName`

Reply:
This is funny. It's actually a bug in TS. 
https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/main/baselines/dom.generated.d.ts#L17613 - these permissions are simply missing that's why casting fails. If they were present in the type definition the simplest solution would be to use `const descriptors = ['camera', 'microphone'] as const` - https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-inference 

TS has some issues with these types for some reason. That kind of issues happen from time to time: https://github.com/microsoft/TypeScript/issues/33923

Reply:
Nice find! I've pushed an update adding a comment with a link to an issue and I'm now casting the name rather than using `Extract` (I think it's a tiny bit cleaner). I was tempted to create a PR to fix the issue but it looks like camera & microphone are considered [at risk](https://w3c.github.io/permissions/#dom-permissionname-camera)